code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectMemberResourceManager.java

```diff
@@ -0,0 +1,123 @@
+import io.cattle.platform.iaas.api.auth.dao.AuthDao;
+import io.cattle.platform.iaas.api.auth.github.resource.Member;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.ListOptions;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.util.*;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class ProjectMemberResourceManager extends AbstractObjectResourceManager {
+
+    @Inject
+    JsonMapper jsonMapper;
+    @Inject
+    AuthDao authDao;
+    @Inject
+    GenericResourceDao resourceDao;
+    @Inject
+    ObjectManager objectManager;
+    @Inject
+    ObjectProcessManager objectProcessManager;
+
+    private List<ExternalIdHandler> externalIdHandlers;
+
+    @Override
+    protected Object removeFromStore(String type, String id, Object obj, ApiRequest request) {
+        return null;
+    }
+
+    @Override
+    protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Object, Object> criteria, ListOptions options) {
+        Policy policy = (Policy) ApiContext.getContext().getPolicy();
+        String id = (String) criteria.get("id");
+        if (StringUtils.isNotEmpty(id)) {
+            ProjectMember projectMember = authDao.getProjectMember(Long.valueOf(id));
+            if (!authDao.hasAccessToProject(projectMember.getProjectId(), policy.getAccountId(),
+                    policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS), policy.getExternalIds())) {
+                throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+            }
+            policy.grantObjectAccess(projectMember);
+            return Arrays.asList(projectMember);
+        }
+        id = (String) criteria.get("projectId");
+        if (StringUtils.isNotEmpty(id)) {
+            List<? extends ProjectMember> members = authDao.getActiveProjectMembers(Long.valueOf(id));
+            for (ProjectMember member: members){
+                policy.grantObjectAccess(member);
+            }
+            return members;
+        }
+        List<? extends ProjectMember> members = authDao.getActiveProjectMembers(policy.getAccountId());
+        for (ProjectMember member: members){
+            policy.grantObjectAccess(member);
+        }
+        return members;
+    }
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { "projectMember" };
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[0];
+    }
+
+    public List<ProjectMember> setMembers(Account project, List<Map<String, String>> members) {
```

Comment:
Short story: I don't want to fight for it that much so I'm fine with keeping it where it is. :-)

In case you were curious, this was my line of thinking.  My initial reaction was that it shouldn't even be in either classes since both inherit from AbstractObjectResourceManager which seemed to handle things directly related to the public API rather than helper functions that we just use internally in our handlers.  For these helper functions, it seems like we usually just create a different manager or service class with its own interface.  When I checked to see where it was used, I saw it was used by ProjectResourceManager which also inherits from AbstractObjectResourceManager.  If it wasn't used elsewhere we could've moved it there as a private function.  But since it is used by SetProjectMembersActionHandler, we have to expose it them.
Also, to me, logically, ProjectResourceManager manages "project" resources.  Having a "setMembers()" function made sense since we're "setting the members" on a "project" resource.